odxtools/diaglayer.py

```diff
@@ -462,12 +461,17 @@ def decode_response(
 
     def get_communication_parameter(self,
                                     name: str,
+                                    functional: Optional[bool] = None,
                                     protocol_name: Optional[str] = None) \
             -> Optional[CommunicationParameterRef]:
 
         cps = [cp for cp in self.communication_parameters if cp.short_name == name]
+
+        if functional is not None:
+            cps = [cp for cp in cps if cp.is_functional == functional]
         if protocol_name:
-            cps = [cp for cp in cps if cp.protocol_sn_ref in (None, protocol_name)]
+            cps = [cp for cp in cps if cp.protocol_sn_ref == protocol_name]
```

Comment:
the idea of the `cp.protocol_sn_ref in (None, protocol_name)` statement was that if no protocol SNREF is specified by the comparam ref, the param always applies. maybe I misread the spec here, but if not, you probably need to add the same logic for the `is_functional` logic two lines above...

Reply:
I think you are right regarding the protoco, I will rollback that change. However, since `is_functional` is truely boolean, we do not need to adapt the logic above